Improve error message for obtaining a JDBC connection

When the application tries to access a JDBC connection, we throw the following error:

```
Caused by: java.sql.SQLException: Not using JDBC
```

This happens in [NoJdbcConnectionProvider#getConnection](https://github.com/hibernate/hibernate-reactive/blob/3d6cc0fb74c729788132ce35da948705ac03ab2d/hibernate-reactive-core/src/main/java/org/hibernate/reactive/provider/service/NoJdbcConnectionProvider.java#L25).

When this happens, it means that we haven't tested a particular mapping or configuration. In a perfect world, a user should never see this error.

We should think of a better error message to explain the user that their use case is currently non supported.
And it makes sense to add the error in the logger.

--- a/hibernate-reactive-core/src/main/java/org/hibernate/reactive/logging/impl/Log.java
+++ b/hibernate-reactive-core/src/main/java/org/hibernate/reactive/logging/impl/Log.java
@@ -7,6 +7,7 @@ package org.hibernate.reactive.logging.impl;
 
 
 
+import java.sql.SQLException;
 import java.sql.SQLWarning;
 
 import jakarta.persistence.PersistenceException;
@@ -265,6 +266,9 @@ public interface Log extends BasicLogger {
 
 	@Message(id = 83, value = "Unexpected request of a non reactive connection")
 	HibernateException unexpectedConnectionRequest();
+
+	@Message(id = 84, value = "The application requested a JDBC connection, but Hibernate Reactive doesn't use JDBC. This could be caused by a bug or the use of an unsupported feature in Hibernate Reactive")
+	SQLException notUsingJdbc();
 
 	// Same method that exists in CoreMessageLogger
 	@LogMessage(level = WARN)

--- a/hibernate-reactive-core/src/main/java/org/hibernate/reactive/provider/service/NoJdbcConnectionProvider.java
+++ b/hibernate-reactive-core/src/main/java/org/hibernate/reactive/provider/service/NoJdbcConnectionProvider.java
@@ -6,9 +6,13 @@
 package org.hibernate.reactive.provider.service;
 
 import org.hibernate.engine.jdbc.connections.spi.ConnectionProvider;
+import org.hibernate.reactive.logging.impl.Log;
 
 import java.sql.Connection;
 import java.sql.SQLException;
+
+import static java.lang.invoke.MethodHandles.lookup;
+import static org.hibernate.reactive.logging.impl.LoggerFactory.make;
 
 /**
  * A dummy Hibernate {@link ConnectionProvider} throws an
@@ -17,12 +21,13 @@ import java.sql.SQLException;
  * @author Gavin King
  */
 public class NoJdbcConnectionProvider implements ConnectionProvider {
+	private static final Log LOG = make( Log.class, lookup() );
 
 	public static final NoJdbcConnectionProvider INSTANCE = new NoJdbcConnectionProvider();
 
 	@Override
 	public Connection getConnection() throws SQLException {
-		throw new SQLException( "Not using JDBC" );
+		throw LOG.notUsingJdbc();
 	}
 
 	@Override

